src/reaktoro_pse/reaktoro_block.py

```diff
@@ -0,0 +1,952 @@
+                        )
+
+    def build_rkt_jacobian(self, block):
+        """this will build rkt jacboian on specified block.
+        The keyword arguments are for automatic configuration of speciation and property blocks
+
+        Keywords:
+        block -- pyomo block to build the model on
+        """
+        """ config outputs """
+        block.rkt_jacobian = ReaktoroJacobianSpec(block.rkt_state, block.rkt_outputs)
+        block.rkt_jacobian.configure_numerical_jacobian(
+            jacobian_type=self.config.numerical_jac_type,
+            order=self.config.numerical_jac_order,
+            step_size=self.config.numerical_jac_step,
+        )
+
+    def build_rkt_solver(self, block, speciation_block=False):
+        """this will build rkt outputs specified block.
+        The keyword arguments are for automatic configuration of speciation and property blocks
+
+        Keywords:
+        block -- pyomo block to build the model on
+        speciation_block -- sets to build speciation block which by default does not
+            add mineral or gas phases (unless configured to do so), and only outputs speciesAmount from the rktModel
+        """
+
+        """config solver """
+        name = str(self)
+        if speciation_block:
+            name = f"{name}_speciation_block"
+        block.rktSolver = ReaktoroSolver(
+            block.rkt_state,
+            block.rkt_inputs,
+            block.rkt_outputs,
+            block.rkt_jacobian,
+            block_name=name,
+        )
+        if speciation_block:
+            presolve = self.config.presolve_speciation_block
+        else:
+            presolve = self.config.presolve_property_block
+        block.rktSolver.set_solver_options(
+            tolerance=self.config.tolerance,
+            epsilon=self.config.epsilon,
+            presolve=presolve,
+            presolve_tolerance=self.config.presolve_tolerance,
+            presolve_epsilon=self.config.presolve_epsilon,
+            max_iters=self.config.max_iters,
+            presolve_max_iters=self.config.presolve_max_iters,
+            hessian_type=self.config.hessian_type,
+        )
+
+    def build_gray_box(self, block):
+        """this will build rkt outputs specified block.
+        The keyword arguments are for automatic configuration of speciation and property blocks
+
+        Keywords:
+        block -- pyomo block to build the model on
+        """
+        """ build block"""
+        scaling = self.config.jacobian_user_scaling
+        scaling_type = self.config.jacobian_scaling_type
+        block.rkt_block_builder = ReaktoroBlockBuilder(
+            block, block.rktSolver, build_on_init=False
+        )
+        block.rkt_block_builder.configure_jacobian_scaling(
+            jacobian_scaling_type=scaling_type, user_scaling=scaling
+        )
+        block.rkt_block_builder.build_reaktoro_block()
+
+    def display_jacobian_outputs(self):
```

Comment:
Rather than a logger here, I would suggest allow the user to provide a stream object (generally a `StringIO`), and if not defaulting to `stdout`. This makes i ta bit easier to handle and test the output.

Also, looking at these I wonder if most of the code could be moved to the jacobian object and this method just be a wrapper around that (sort of like how it already calls `rkt_jacobian.display_jacobian_output_types`, but move all the code to there).

Reply:
Yeah that would make sense. I will look into it.